apps/lib/engine.c

```diff
@@ -162,6 +163,14 @@ int get_legacy_pkey_id(OSSL_LIB_CTX *libctx, const char *algname, ENGINE *e)
     EVP_PKEY_asn1_get0_info(&pkey_id, NULL, NULL, NULL, NULL, ameth);
 
     return pkey_id;
+#else
+    EVP_KEYMGMT *keymgmt = EVP_KEYMGMT_fetch(NULL, algname, NULL);
+    if (keymgmt != NULL) {
+        EVP_KEYMGMT_free(keymgmt);
+        return OBJ_txt2nid(algname);
+    }
+    return NID_undef;
```

Comment:
I do not think this is right as this will also set NIDs to algorithms that do not have them yet. I would simply return NID_undef. Which will likely trigger further changes of course.